nx-dev/ui-courses/src/lib/lesson-player.tsx

```diff
@@ -0,0 +1,220 @@
+              }
+            )}
+            prefetch={false}
+          >
+            <span
+              className={cx('inline-block min-w-[2rem] flex-shrink-0 text-sm', {
+                'text-blue-600 dark:text-blue-400': item.id === lesson.id,
+                'text-slate-400 dark:text-slate-500': item.id !== lesson.id,
+              })}
+            >
+              {(index + 1).toString()}
+            </span>
+            <span className="flex-1 text-sm leading-normal">
+              <span
+                className={cx('', {
+                  'font-semibold': item.id === lesson.id,
+                  'text-slate-700 dark:text-slate-300': item.id !== lesson.id,
+                })}
+              >
+                {item.title}
+              </span>
+            </span>
+            {item.duration && (
+              <span className="ml-2 flex items-center gap-1 text-xs text-slate-400 dark:text-slate-500">
+                <svg
+                  className="h-3 w-3"
+                  fill="none"
+                  viewBox="0 0 24 24"
+                  stroke="currentColor"
+                >
+                  <path
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                    strokeWidth={2}
+                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
+                  />
+                </svg>
+                {formatDuration(item.duration)}
+              </span>
+            )}
+          </Link>
+        </li>
+      ))}
+    </ul>
+  );
+
+  return (
+    <>
+      <style jsx global>
+        {customStyles}
+      </style>
+      <Header />
+      <div className="flex h-screen w-full pt-20">
+        {/* Left Panel - Course Info & Lessons (Hidden on mobile) */}
+        <div className="hidden lg:flex lg:w-80 lg:flex-none lg:flex-col lg:border-r lg:border-slate-200 lg:dark:border-slate-700">
+          <div className="border-b border-slate-200 p-6 dark:border-slate-700">
+            <div className="text-sm font-medium text-slate-500 dark:text-slate-400">
+              Course by {course.authors[0].name}
+            </div>
+            <Link
+              href={`/courses/${course.id}`}
+              className="mt-1 block text-xl font-bold text-slate-900 hover:text-blue-600 dark:text-white dark:hover:text-blue-400"
+            >
+              {course.title}
+            </Link>
+          </div>
+          <div className="flex-1 overflow-y-auto py-4">
+            <LessonsList />
+          </div>
+        </div>
+
+        {/* Center/Right Panel - Video & Content */}
+        <div className="wide:w-[45%] wide:flex-none wide:overflow-y-visible wide:border-r wide:border-slate-200 wide:dark:border-slate-700 flex flex-1 flex-col overflow-y-auto">
+          {/* Video Section */}
+          <div className="flex-none border-b border-slate-200 dark:border-slate-700">
+            {lesson.videoUrl ? (
+              <div className="aspect-video">
+                <YouTube
```

Comment:
Is there a way to disable related video at the end of the cource video? Is there a way we can show a CTA to go to the next lesson when the video is stopped? Or we always show a CTA under the video?

Reply:
Yeah that's planned for an upcoming PR. I saw it as well and looked into it. YT has an API you can leverage that allows you to get information about the watch progress etc, so you could add an overlay towards the end with a button that leads you to the next lesson. Def want to add this, but wanted to keep this first push minimal and just get it out.

Coded most of it literally on the flight back from MW, so I didn't want to spend too much time on it and then incrementally improve it